java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java

```diff
@@ -0,0 +1,115 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.check.Rule;
+import org.sonar.java.JavaVersionAwareVisitor;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaVersion;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.IfStatementTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import java.util.List;
+import java.util.Optional;
+
+@Rule(key = "S2196")
+public class SwitchInsteadOfIfSequenceCheck extends IssuableSubscriptionVisitor implements JavaVersionAwareVisitor {
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return ImmutableList.of(Tree.Kind.IF_STATEMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if(tree.parent().is(Tree.Kind.IF_STATEMENT)) {
+      IfStatementTree parentIf = (IfStatementTree) tree.parent();
+      if(tree.equals(parentIf.elseStatement())) {
+        // don't double count nested ifs
+        return;
+      }
+    }
+    IfStatementTree ifStatementTree = (IfStatementTree) tree;
+    int level = 1;
+    while (ifStatementTree.elseKeyword() != null && ifStatementTree.elseStatement().is(Tree.Kind.IF_STATEMENT)) {
+      level++;
+      if(!sameEqualCondition(ifStatementTree.condition(), ((IfStatementTree) ifStatementTree.elseStatement()).condition())) {
+        level = 0;
+        break;
+      }
+      ifStatementTree = (IfStatementTree) ifStatementTree.elseStatement();
+    }
+    if(level > 2) {
+      reportIssue(tree, "Convert this \"if/else if\" structure into a \"switch\".");
+    }
+
+  }
+
+  private static boolean sameEqualCondition(ExpressionTree firstIfCondition, ExpressionTree secondIfCondition) {
+    Optional<EqualsOperands> equalsOperandsFirst = getEqualMethodInvocationOperands(firstIfCondition);
+    Optional<EqualsOperands> equalsOperandsSecond = getEqualMethodInvocationOperands(secondIfCondition);
+    return equalsOperandsFirst.isPresent()
+    && equalsOperandsSecond.isPresent()
+    && equalsOperandsFirst.get().identifier.symbol().equals(equalsOperandsSecond.get().identifier.symbol());
+  }
+
+  private static Optional<EqualsOperands> getEqualMethodInvocationOperands(ExpressionTree expressionTree) {
+    if(!expressionTree.is(Tree.Kind.METHOD_INVOCATION)) {
+      return Optional.empty();
```

Comment:
given : https://jira.sonarsource.com/browse/RSPEC-1698 I doubt we should detect this case...